eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/FluentMethodNameCheck.java

```diff
@@ -0,0 +1,76 @@
+package com.azure.tools.checkstyle.checks;
+
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+/**
+ * Model Class Method:
+ *  Fluent Methods: All methods that return an instance of the class, and with one parameter.
+ *  The method name should not start with {@code avoidStartWord}.
+ */
+public class FluentMethodNameCheck extends AbstractCheck {
+
+    private static final String FLUENT_METHOD_ERR = "Fluent Method name should not start with keyword %s.";
+
+    // Specifies valid identifier: default start word is 'with'
+    private String avoidStartWord = "with";
+
+    private String className;
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {
+            TokenTypes.CLASS_DEF,
+            TokenTypes.METHOD_DEF
+        };
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+
+        switch (ast.getType()) {
+            case TokenTypes.CLASS_DEF:
+                className = ast.findFirstToken(TokenTypes.IDENT).getText();
```

Comment:
How do you know this class is a "model" class? Maybe you can check the package name for this class.

Reply:
Based on Jonathon's suggestion. Will not add a check for determining a model class